launch_yaml/launch_yaml/entity.py

```diff
@@ -0,0 +1,103 @@
+# Copyright 2019 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Module for YAML Entity class."""
+
+from typing import List
+from typing import Optional
+from typing import Text
+from typing import Tuple
+from typing import Union
+
+from launch_frontend import Entity as BaseEntity
+from launch_frontend.type_utils import check_type
+
+
+class Entity(BaseEntity):
+    """Single item in the intermediate YAML front_end representation."""
+
+    def __init__(
+        self,
+        element: dict,
+        *,
+        type_name: Optional[Text] = None,
+        parent: 'Entity' = None
+    ) -> Text:
+        """Construnctor."""
+        if type_name is None:
+            if len(element) != 1:
+                raise RuntimeError('Expected a len 1 dictionary')
+            self.__type_name = list(element.keys())[0]
+            self.__element = element[self.__type_name]
```

Comment:
@ivanpauno mind to explain the logic here a bit? Is this to cope with the root dictionary?

Reply:
Yeah this is a little confusing ...
- The usual loading is with `type_name=None`. Each yaml launch description is expected to only have one root element, if not an error is raised. Then, I store the (only) key as the name, and the value as the element tree. Having this constructor is also a good idea when using the `children` property, if one of the children is ill formed an error will be raised.
- The "unusual" loading, is with `typename='somestringhere'`. When using `get_attr('sth', types='list[Entity]')`, you have a dict with a list inside. Each of the list elements has to be loaded with the same type_name (the key of the dict).

The real solution would be to always specify the `type_name` argument, and have some formal checking before to check if the yaml file is ill formed or not.

Reply:
I finally changed it, enforcing passing `type_name`.